File: Java-SpringBoot/BE-02-DB/java-21-springboot-4.0/src/main/java/com/flyrank/taskservice/dto/TaskUpdateRequest.java
package com.flyrank.taskservice.dto;

import io.swagger.v3.oas.annotations.media.Schema;

@Schema(description = "Payload accepted by PATCH /tasks/{id} for partial update")
public record TaskUpdateRequest(
        @Schema(description = "Updated task title (optional)", example = "Partially updated title")
        String title,

        @Schema(description = "Updated completion status (optional)", example = "true")
        Boolean done
) {}

File: Java-SpringBoot/BE-02-DB/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/taskservice/dto/TaskUpdateRequest.kt
package com.flyrank.taskservice.dto

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Payload accepted by PATCH /tasks/{id} for partial update")
data class TaskUpdateRequest(
    @field:Schema(description = "Updated task title (optional)", example = "Partially updated title")
    val title: String? = null,

    @field:Schema(description = "Updated completion status (optional)", example = "true")
    val done: Boolean? = null
)
